src/common.rs

```diff
@@ -343,6 +369,10 @@ impl ScaledFloat {
     pub fn into_value(self) -> f32 {
         Self::reverse(self.0) as f32
     }
+
+    pub fn encode<W: Write>(self, w: &mut W) -> encoder::Result<()> {
+        encoder::write_chunk(w, chunk::gAMA, &self.into_scaled().to_be_bytes())
+    }
```

Comment:
`ScaledFloat` is used in different chunks, exposing `encode` such that it encodes it as a gama chunk could be confusing.
```suggestion
    pub(crate) fn encode_gama<W: Write>(self, w: &mut W) -> encoder::Result<()> {
        encoder::write_chunk(w, chunk::gAMA, &self.into_scaled().to_be_bytes())
    }
```